extensions/reg-core-3.3.md

```diff
@@ -0,0 +1,209 @@
+authority layer of the network.  Further, we believe it is helpful to client authors, so that
+they may provide guided account creation and validation for their users, regardless of network
+configuration.
+
+## The `REG` command
+
+The `REG` command is the main command used by the account registration subsystem.  It provides
+several subcommands listed in the `REGCOMMANDS` ISUPPORT type.  The core sub-commands are listed
+below:
+
+## The `REG CREATE` sub-command
+
+The `REG CREATE` command signals the intent of a client to register an account in the network's
+authentication layer.  It is similar to current methods of signalling that intent.
+
+A `REG CREATE` command consists of the following format:
+
+`REG CREATE <accountname> <callback> <cred_type> :<credential>`
+
+A passphrase `credential` MAY have spaces in it.
+
+The `callback` field designates an opaque value which indicates where a verification code, if any,
+will be sent.  The callback field is implementation-specific, with namespaces being listed using the
+`REGCALLBACKS` ISUPPORT token.  An invalid callback parameter should result in the `ERR_REG_INVALID_CALLBACK`
+error.  In the event that a callback is not provided, the client SHOULD send `*` to indicate that
+they are not providing a callback resource.
+
+The `cred_type` field designates a value which indicates the type of supplied credential.  The accepted
+values of the `cred_type` field is implementation-specific, with credential types being listed using the
+`REGCREDTYPES` ISUPPORT token.  An invalid credential type parameter should result in the `ERR_REG_INVALID_CRED_TYPE`
+error.  The client MUST supply a `cred_type` value.
+
+The IRC server MAY forward the `REG CREATE` command to a central authority, or process it locally, however
+the authentication layer SHOULD NOT send any legacy messages in reply to a `REG CREATE` command.
+
+Upon success, the IRC server MUST send the `RPL_REGISTRATION_SUCCESS` numeric, which looks like:
+
+    :<server> 920 <user_nickname> <accountname> :Account created
+
+If the server requires a verification token, it MUST also reply with the `RPL_REG_VERIFICATION_REQUIRED`
+numeric, which looks like:
+
+    :<server> 927 <user_nickname> <accountname> <callback> :A verification token was sent
+
+Upon error, the IRC server MUST send an error code that is relevant.  We suggest these numerics:
+
+`ERR_ACCOUNT_ALREADY_EXISTS`:
+    `:<server> 921 <user_nickname> <accountname> :Account already exists`
+`ERR_REG_UNSPECIFIED_ERROR`:
+    `:<server> 922 <user_nickname> <accountname> :<descriptive_text>`
+`ERR_REG_INVALID_CALLBACK`:
+    `:<server> 929 <user_nickname> <accountname> <callback> :Callback token is invalid`
+`ERR_REG_INVALID_CRED_TYPE`:
+    `:<server> 928 <user_nickname> <accountname> <cred_type> :Credential type is invalid`
+`ERR_REG_UNAVAILABLE`:
+    `:<server> 440 <user_nickname> <subcommand> :Authentication services are currently unavailable`
```

Comment:
The rendered markdown of this section joins all the lines as one. Same thing with the numerics list of the next section.

Checked the other specs with lists of numerics, all of them do it differently, so I guess anything that is readable will do.